### DISCUSSION TITLE
How do I use PerfView to Collect for a 32-bit app specifically for the System.Data.1 provider

### DISCUSSION THREAD

**Discussion by mdell-seradex** [microsoft/perfview#1813] [Q&A]
I ran the following command to register the providers into the WMI registry, but I am unable to get any events.  I suspect it is because I am attempting to profile a 32-bit process.
`mofcomp.exe C:\Windows\Microsoft.NET\Framework64\v4.0.30319\adonetdiag.mof`

I have tried to manually perform tracing using `logman` and managed to get it to work for a while last week, but I cannot get it to work anymore.  Logman will only generate the basic 8kb file.
I am unsure what I am doing wrong.

I feel like using Perfmon would be an easier way to gather the required data, but it is not recording any events from the selected additional provider.

I appreciate any help or feedback that can be provided.

**Comment by mdell-seradex** [comment]
I discovered the issue being that the registry entry need to tell the apps which should enable tracing is

1. Still needed, and
2. Does not support environment variables, unlike what the documentation at [Data Access Tracing in SQL Server 2008 - Creating the Trace DLL Registry Entry (ADO.NET 2.0)](https://learn.microsoft.com/en-us/previous-versions/sql/sql-server-2008/cc765421(v=sql.100)#:~:text=ADO.NET%202.0-,Value,-%3D%22%25SYSTEMROOT%25%5CMicrosoft.NET) demonstrates.

Other than that it seems to be able to successfully trace 32-bit apps.

I don't know if you could provide some info on that in the app, even if just to say that others may require additional setup. Something more specific for this would be great, but I can understand if you don't/can't.

**Reply by brianrob** [reply]
@mdell-seradex, I'm glad you got things working.  To make sure I understand, are you saying you needed to edit the registry?

**Reply by mdell-seradex** [reply]
The documentation at that link ([Data Access Tracing in SQL Server 2008 - Creating the Trace DLL Registry Entry (ADO.NET 2.0)](https://learn.microsoft.com/en-us/previous-versions/sql/sql-server-2008/cc765421(v=sql.100)#:~:text=ADO.NET%202.0-,Value,-%3D%22%25SYSTEMROOT%25%5CMicrosoft.NET)) states that I need to add a registry entry to enable ETW for the ADO type providers such as `System.Data.1`.  This is one of the providers in `adonetdiag.mof` that is added to the PC's WMI database using that `mofcomp` command.
That documentation incorrectly shows using an environment variable in that registry entry `%SYSTEMROOT%`, however, I have discovered via my testing that it is not supported.  The actual path MUST be specified.

**Reply by brianrob** [reply]
I see.  This looks like reasonably old content, and so I don't see a way to edit it via GitHub, which is how newer content is edited.  If you scroll to the bottom of the page, there is a feedback link, which will allow you to send an e-mail.  I would recommend reaching out there, and hopefully someone can help to get the content fixed.

**Reply by mdell-seradex** [reply]
I tried that.  It failed.  I think it no longer exists.
Note that while I found that article first, I was able to trace it to active content.
This active page: [https://learn.microsoft.com/en-us/dotnet/framework/data/adonet/data-tracing (see Tracing Data Access)](https://learn.microsoft.com/en-us/dotnet/framework/data/adonet/data-tracing#:~:text=For%20more%20information%20about%20setting%20and%20configuring%20managed%20tracing%20in%20ADO.NET%2C%20see%20Tracing%20Data%20Access.)
Points at this page: [https://learn.microsoft.com/en-us/previous-versions/sql/sql-server-2012/hh880086(v=msdn.10) (For information on how to do data access tracing)](https://learn.microsoft.com/en-us/previous-versions/sql/sql-server-2012/hh880086(v=msdn.10)#:~:text=For%20information%20on%20how%20to%20do%20data%20access%20tracing%2C%20see%3A)
Which points at the one above: https://learn.microsoft.com/en-us/previous-versions/sql/sql-server-2008/cc765421(v=sql.100)

I did try submitting a Github issue for the active content, but nothing as yet.

**Reply by brianrob** [reply]
Gotcha.  In that case, I would recommend filing an issue here in the dotnet/docs repo: https://github.com/dotnet/docs/blob/main/docs/framework/data/adonet/data-tracing.md.  They're likely to be the best folks to help with this.

**Reply by mdell-seradex** [reply]
I just figured since PerfView can use those providers, I thought some mention by it about the potential for additional steps for the additional providers may be a good idea.

Thank you for the response & interest in my discussion post.

**Reply by brianrob** [reply]
I see.  I think if we were going to put a note somewhere in PerfView, it would be in the Users Guide.  Does this provider get turned on by any of PerfView's existing configurations?  I did a quick search, but I don't see it.  If we don't, then I'm not sure where we'd put it.  If we do, then we could potentially put something with the command line argument.

**Reply by mdell-seradex** [reply]
This is how I added the `System.Data.1` Provider.  I had already run that `mofcomp.exe` command on my PC.  I suspect that it won't show on yours until you run it.
My understanding it that both the `mofcomp.exe` and the `adonetdiag.mof` are deployed with Windows.  I was able to run that command on a client's PC and I didn't have to install anything extra to do it.  Nor, as far as I am aware, did they have any development tools installed.
![image](https://user-images.githubusercontent.com/40300659/221721156-2fd2ea0c-6d31-4c36-8e61-a6bd6fb58cc4.png)

**Reply by brianrob** [reply]
I see.  In that case, we shouldn't add anything specific in PerfView.  This data comes from the operating system, so any registered provider can show up here.  I think the best course of action here would be to modify the existing documentation for the provider itself, rather than to put something specific into PerfView.

**Reply by mdell-seradex** [reply]
I understand not adding anything specific, which is why I suggested as well the possiblity of a general note `to say that others may require additional setup`.
Perhaps something like "Providers added via the 'ProviderBrowser' may require additional setup such as registry keys.  Please refer to their documentation for more information."

**Comment by brianrob** [comment]
Ah, gotcha.  What do you think about putting a line in the users guide here: https://github.com/microsoft/perfview/blob/main/src/PerfView/SupportFiles/UsersGuide.htm#L4458

If that sounds reasonable, would you be up for submitting a PR?

**Reply by mdell-seradex** [reply]
I've only done that a couple of times on Github, but I can give it a shot.  Both times my changes were re-written 😶.
